### PR TITLE
Initiate download of submission report PDF

### LINKED ISSUE #4488: Force downloading of PDF from confirmation page, instead of letting the browser open.
### Thema / Theme

Frontend

### Omschrijving / Description

Browsers are very eager to open pdf's for their users, but this behavior does not really fit with the kinds of PDF's that OF generates, which are definitely more of the kind you would want to actually save on your disk.

Since the user loses the ability to download the pdf again after the session is gone, the 
"Download PDF" should force a actual download of the PDF via the "download" attribute which seems to be supported by all [modern browsers.](https://caniuse.com/mdn-html_elements_a_download)

- [x] Set up proper URL without exposing API endpoint details
- [x] Add Content-Disposition headers to force download

### PR DESCRIPTION
**Pull request by sergei-maertens**
Closes #4488

**Changes**

* Refactored to use the PrivateMediaView generic CBV
* URL moved to non-API url
* Deleted the API-related code that is now obsolete

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

File: src/openforms/submissions/views.py [hunk 1]
```diff
@@ -192,6 +194,46 @@ def get_form_resume_url(self, submission: Submission) -> str:
         )
 
 
+class DownloadSubmissionReportView(PrivateMediaView):
+    """
+    Download the PDF report containing the submission data.
+
+    This URL requires a token which is tied to the submission from the session. The
+    token automatically expired after
```
Comment (vaszig) [inline src/openforms/submissions/views.py:202]:
```suggestion
    token automatically expires after
```

File: src/openforms/submissions/views.py [hunk 2]
```diff
@@ -192,6 +194,46 @@ def get_form_resume_url(self, submission: Submission) -> str:
...
+            "mimetype": "application/pdf",
+        }
+        return opts
+
+    def get(self, request: HttpRequest, *args, **kwargs):
+        self.object = self.get_object()
+
+        response = super().get(request, *args, **kwargs)
+
+        self.object.last_accessed = timezone.now()
```
Comment (vaszig) [inline src/openforms/submissions/views.py:232]: Do we need local time here, like `datetime_in_amsterdam` ?And basically is there a standard for when we need the local time?

**Review by vaszig — APPROVED** [review]
Only one small typo.

File: src/openforms/submissions/views.py (hunk 2 above)
Comment (Viicos) [reply src/openforms/submissions/views.py:232]: I think the difference between using `timezone.now` vs `datetime_in_amsterdam` is whether you want the datetime displayed (i.e. string formatted or not). `last_accessed` is a db field and they are stored with UTC timezones so it's good to use `timezone.now()`

Sergei might confirm

File: src/openforms/submissions/views.py [hunk 3]
```diff
@@ -192,6 +194,46 @@ def get_form_resume_url(self, submission: Submission) -> str:
...
+    def has_permission(self):
+        report: SubmissionReport = self.get_object()
+        token = self.kwargs["token"]
+        return submission_report_token_generator.check_token(report, token)
+
+    def get_sendfile_opts(self) -> dict:
+        path = Path(self.object.content.name)
+        opts = {
+            "attachment": True,
+            "attachment_filename": path.name,
```
Comment (Viicos) [inline src/openforms/submissions/views.py:222]: What is the reason to go through `Path(self.object.content.name).name` and not just `self.object.content.name`?

**Review by Viicos — APPROVED** [review]
I guess the SDK needs to be updated with the new API endpoint? Or the URL is sent by the backend?

File: src/openforms/submissions/views.py (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/submissions/views.py:232]: no, django converts everything to UTC anyway when it's being saved in the DB. The `datetime_in_amsterdam` is only needed in edge cases:

* presentation to the end-use, usually this is handled already in the django template layer automatically
* when interfacing with other systems that don't use ISO-8601 datetimes and expect naive, local times

So usually these are external factors outside of our control.

**Comment by sergei-maertens** [comment]
> I guess the SDK needs to be updated with the new API endpoint? Or the URL is sent by the backend?

URL is sent by the backend, no updates needed. HATEOAS baby :tada:

File: src/openforms/submissions/views.py (hunk 3 above)
Comment (sergei-maertens) [reply src/openforms/submissions/views.py:222]: the former only returns `[basename].[extension]`, the latter returns `some/subpath/from/upload/to/[basename].[extension]`

**Pull request merged**